inject-generator/src/main/java/io/avaje/inject/generator/MethodReader.java

```diff
@@ -179,7 +181,21 @@ void addImports(Set<String> importTypes) {
   }
 
   Set<GenericType> getGenericTypes() {
-    return typeReader == null ? Collections.emptySet() : typeReader.getGenericTypes();
+    return typeReader != null ? typeReader.getGenericTypes() : Collections.emptySet();
+  }
+
+  void factoryImports(Set<String> importTypes) {
+    Set<GenericType> genericTypes = getGenericTypes();
+    if (!genericTypes.isEmpty()) {
+      importTypes.add(Constants.TYPE);
+      importTypes.add(Constants.GENERICTYPE);
+      importTypes.addAll(genericTypes
+        .stream()
+        .flatMap(g-> Stream.concat(
+          Stream.of(g.getMainType()),
+          g.getParams().stream().map(GenericType::getMainType)))
```

Comment:
Just noting that GenericType.addImports() does this for us.  I added a commit to do this part so I think I'll merge this in and then trim out the bits that are now duplicated.

Good stuff !!!

Reply:
Sounds good 👍